xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp

```diff
@@ -2267,53 +2228,21 @@ void CAMLCodec::SetVideoSaturation(const int saturation)
   SysfsUtils::SetInt("/sys/class/video/saturation", saturation);
 }
 
-bool CAMLCodec::SetVideo3dMode(const int mode3d)
-{
-  bool result = true;
-  if (SysfsUtils::Has("/sys/class/ppmgr/ppmgr_3d_mode"))
-  {
-    CLog::Log(LOGDEBUG, "CAMLCodec::SetVideo3dMode:mode3d(0x%x)", mode3d);
-    SysfsUtils::SetInt("/sys/class/ppmgr/ppmgr_3d_mode", mode3d);
-  }
-  else
-  {
-    CLog::Log(LOGINFO, "CAMLCodec::SetVideo3dMode: ppmgr_3d support not found in kernel.");
-    result = false;
-  }
-  return result;
-}
-
-std::string CAMLCodec::GetStereoMode()
+std::string CAMLCodec::GetVideoStereoMode()
 {
   std::string stereoMode;
 
-  //Get Decoder Stereo mode
-  int decoder_sm(VPP_3D_MODE_NULL);
-  aml_ioctl_get(am_private->vcodec.handle, AMSTREAM_IOC_GET_3D_TYPE, (unsigned long)&decoder_sm);
-
-  switch (decoder_sm)
+  switch (m_processInfo.GetVideoSettings().m_StereoMode)
```

Comment:
Where do we get the information if 3d yes/no from stream?
This part here was changed not long time ago after discussion with @FernetMenta because the information from stream is leading here.

Reply:
If I understand correctly the info about 3D type is coming either from metadata or from name of the file being played. For decoder SBS and TAB videos are just plain flat videos divided into two parts, there's no any info about 3D in the stream. It is a job of 3D TV to make final 3D picture.

Reply:
under internal discussion currently....

Reply:
videoSettings is something different. Via VideoSettings you can enforce SBS or TAB mode in case all other detection ways failed. That was at least the initial aim when this was added by elupus (IIRC)

Reply:
@codesnake are you sure that VideoSettings shold have higher priority than the detected 3d mode?

Reply:
This method is used only for 3D to 2D conversion to figure out what view of the video to use for 2D display - top for TAB or left for SBS.

Reply:
@da-anda And that's exactly the way my code handles it - use VideoSettings if user forced 3D mode there, otherwise use hints.stereoMode.

Reply:
@peak3d yes, AFAIK VideoSettings override the hints or filename flags, since via video settings you can also flip the eyes. So settings overrule

Reply:
@codesnake we looked over the core code and are quite sure that you can skip `std::string CAMLCodec::GetVideoStereoMode()` and use CDataCacheCode::GetVideoStereoMode instead.

This is what already is done in VideoPlayer, and it's identical to that whats in this function:
https://github.com/xbmc/xbmc/blob/d7aea79779ad880f0232f942768f24cd63205baf/xbmc/cores/VideoPlayer/VideoPlayerVideo.cpp#L686-L701

Reply:
OK, thanks, I will try that.

Reply:
you might want to get the datacacheCore form the service broaker, like `CServiceBroker::GetDataCacheCore().GetVideoStereoMode();`

Reply:
@codesnake DataCacheCode is intended for outside usage, pls use m_processInfo instead.

Reply:
@peak3d So leave it as is?

Reply:
processInfo should also have the correct value, so maybe give it a try with simply using `m_processInfo.GetVideoStereoMode()` instead of your construct?